### PR TITLE
Changes in spellchecker interface

### PR DESCRIPTION
**Pull request by khazhoyan**
see commit message

### CODE REVIEW

File: lib/ace/ext/papeeria/spellchecker.coffee [hunk 1]
```diff
@@ -1,48 +1,46 @@
...
+      @asyncFetchTypos = ->        # {AsyncFetchTypos}
+      @asyncFetchSuggestions = ->  # {AsyncFetchSuggestions}
 
     _fetchTypos: (hash) =>
-      $.getJSON(@typosUrl, null, (typosArray) =>
+      @asyncFetchTypos(@language, (typosArray) =>
         @editor.getSession()._emit("updateSpellcheckingTypos", {typos: typosArray})
         @typosHash = hash
-      ).fail(@onError)
+      )
```
Comment (dbarashev) [inline lib/ace/ext/papeeria/spellchecker.coffee]: failure is supposed to be handled elsewhere?

File: lib/ace/ext/papeeria/spellchecker.coffee [hunk 2]
```diff
@@ -1,48 +1,46 @@
...
     # Update typos hash and refresh list of typos iff hash is different
     # @param {String} typosHash: hash used to check whether the typos list has been changed
     onHashUpdated: (typosHash) =>
       if @typosHash != typosHash
         @_fetchTypos(typosHash)
 
-    # Update spellchecking session
-    # @param {String} typosUrl: url used to fetch typos
-    # @param {String} suggestionsUrl: url used to fetch corrections for a word
-    onSessionUpdated: (typosUrl, suggestionsUrl) =>
```
Comment (dbarashev) [inline lib/ace/ext/papeeria/spellchecker.coffee]: not needed anymore?

File: lib/ace/ext/papeeria/spellchecker.coffee (hunk 1 above)
Comment (khazhoyan) [inline lib/ace/ext/papeeria/spellchecker.coffee]: yup

File: lib/ace/ext/papeeria/spellchecker.coffee (hunk 2 above)
Comment (khazhoyan) [inline lib/ace/ext/papeeria/spellchecker.coffee]: merged with `onSettingsUpdated` since they were called at once pretty much all the time

**Comment by dbarashev** [comment]
LGTM

**Pull request merged**